nifi-nar-bundles/nifi-splunk-bundle/nifi-splunk-processors/src/main/java/org/apache/nifi/processors/splunk/SplunkAPICall.java

```diff
@@ -0,0 +1,250 @@
+    }
+
+    private ServiceArgs getSplunkServiceArgs(final ProcessContext context) {
+        final ServiceArgs splunkServiceArguments = new ServiceArgs();
+
+        splunkServiceArguments.setScheme(context.getProperty(SCHEME).getValue());
+        splunkServiceArguments.setHost(context.getProperty(HOSTNAME).evaluateAttributeExpressions().getValue());
+        splunkServiceArguments.setPort(context.getProperty(PORT).evaluateAttributeExpressions().asInteger());
+
+        if (context.getProperty(OWNER).isSet()) {
+            splunkServiceArguments.setOwner(context.getProperty(OWNER).evaluateAttributeExpressions().getValue());
+        }
+
+        if (context.getProperty(TOKEN).isSet()) {
+            splunkServiceArguments.setToken(context.getProperty(TOKEN).evaluateAttributeExpressions().getValue());
+        }
+
+        if (context.getProperty(USERNAME).isSet()) {
+            splunkServiceArguments.setUsername(context.getProperty(USERNAME).evaluateAttributeExpressions().getValue());
+        }
+
+        if (context.getProperty(PASSWORD).isSet()) {
+            splunkServiceArguments.setPassword(context.getProperty(PASSWORD).getValue());
+        }
+
+        if (HTTPS_SCHEME.equals(context.getProperty(SCHEME).getValue()) && context.getProperty(SECURITY_PROTOCOL).isSet()) {
+            splunkServiceArguments.setSSLSecurityProtocol(SSLSecurityProtocol.valueOf(context.getProperty(SECURITY_PROTOCOL).getValue()));
+        }
+
+        return splunkServiceArguments;
+    }
+
+    protected Service getSplunkService(final ServiceArgs splunkServiceArguments) {
+        return Service.connect(splunkServiceArguments);
+    }
+
+    @OnUnscheduled
+    public void onUnscheduled() {
+        if (splunkService != null) {
+            splunkService.logout();
+            splunkService = null;
+        }
+
+        requestChannel = null;
+        ackIdAttributeName = null;
+        insertedAtAttributeName = null;
+        splunkServiceArguments = null;
+    }
+
+    protected ResponseMessage call(final String endpoint, final RequestMessage request)  {
+        request.getHeader().put(REQUEST_CHANNEL_HEADER_NAME, requestChannel);
+
+        try {
+            return splunkService.send(endpoint, request);
+            //Catch Stale connection exception, reinitialize, and retry
+        } catch (final com.splunk.HttpException e) {
+            getLogger().error("Splunk request status code:" + e.getStatus() + " Retrying the request.");
+            splunkService.logout();
+            splunkService = getSplunkService(splunkServiceArguments);
+            return splunkService.send(endpoint, request);
+        }
+    }
+
+    protected <T> T extractResult(final InputStream responseBody, final Class<T> type) throws IOException {
+        final JsonParser jsonParser = jsonFactory.createParser(responseBody);
+        jsonParser.setCodec(jsonObjectMapper);
+        return jsonParser.readValueAs(type);
```

Comment:
Is there a reason for not using `jsonObjectMapper.readValue(responseBody, type)`?

Reply:
Nice improvement, I was not aware of this